stan/math/prim/fun/chol2inv.hpp

```diff
@@ -20,22 +20,22 @@ namespace math {
  * @throw std::domain_error If the input matrix is not square or
  *  lower triangular
  */
-template <typename T>
-Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> chol2inv(
-    const Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>& L) {
-  check_square("chol2inv", "L", L);
-  check_lower_triangular("chol2inv", "L", L);
+template <typename T, require_eigen_t<T>* = nullptr>
+plain_type_t<T> chol2inv(const T& L) {
+  const Eigen::Ref<const plain_type_t<T>>& L_ref = L;
+  check_square("chol2inv", "L", L_ref);
+  check_lower_triangular("chol2inv", "L", L_ref);
   int K = L.rows();
-  using matrix_t = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>;
+  using matrix_t = plain_type_t<T>;
```

Comment:
typedefs in the rest of the library are generally ```T_```

Reply:
fixed